### ISSUE TITLE
getPurchaseHistory() and getAvailablePurchases() are not resolved / rejected

### ISSUE THREAD

**Issue by GuvanchBayryyyev** [dooboolab/flutter_inapp_purchase#386]
### Version of flutter_inapp_purchase
5.3.1
### Platforms you faced the error (IOS or Android or both?)
both
### Expected behavior
request should be resolved/rejected
### Actual behavior
request not being resolved/rejected
### Tested environment (Emulator? Real Device?)
Emulator and Real Device
### Steps to reproduce the behavior
final history = await _iap.getPurchaseHistory();
final purchases = await _iap.getAvailablePurchases();
### Similar issue
I think it is same issue as in React native [this](https://github.com/dooboolab/react-native-iap/issues/981)

**Comment by marcin03** [comment]
The same problem on iPhone 14 ios 16.0 simulator, but on iPhone 8 ios 12.4 simulator it works

**Comment by maticla** [comment]
Same issue here and it appears to be happening only in sandbox environment. It works on our production build on iOS 15 & iOS 16 but not on the development build in sandbox environment on iOS 16. 

I wrote a race condition hack to get around this:

```
List<PurchasedItem> storePurchases = [];

final result = await Future.any([
  _plugin.getAvailablePurchases(),
  Future.delayed(const Duration(seconds: 5))
]);

result != null ? storePurchases = result : storePurchases = [];
```

But this is still not ideal because you basically cannot retrieve available purchases on iOS 16 at the moment.

**Comment by maticla** [comment]
I have the same issue on iPhone 12 - iOS 16. 

This happens when you are trying to call `getPurchaseHistory()` and `getAvailablePurchases() ` on a development build and you don't are not logged in with a sandbox tester account on your device. Looks like the plugin isn't handling the errors and exceptions thrown by the two of above mentioned methods.

I have added a hack to my initializer, basically a race condition that skips this check if it takes too long.

```
List<PurchasedItem> storePurchases = [];

final result = await Future.any([
  _plugin.getAvailablePurchases(),
  Future.delayed(const Duration(seconds: 5))
]);

result != null ? storePurchases = result : storePurchases = [];
```

I then went to Settings -> App Store -> Scroll down to "Sandbox Account" and enter your credentials. The methods should work fine then.

If I am understanding this issue correctly, this is impossible to happen on release build on AppStore because the user will be always signed in with his/her Apple ID.
